### PR TITLE
Fix formatting, add CI checking for formatting

### PR DESCRIPTION
**Pull request by knickish**

### CODE REVIEW

File: src/serde_bin.rs [hunk 1]
```diff
@@ -390,7 +390,7 @@ where
     fn de_bin(o: &mut usize, d: &[u8]) -> Result<Self, DeBinErr> {
         unsafe {
             let mut to = core::mem::MaybeUninit::<[T; N]>::uninit();
-            let top: *mut T = core::mem::transmute(&mut to);
+            let top: *mut T = &mut to as *mut std::mem::MaybeUninit<[T; N]> as *mut T;
```
Comment (knickish) [inline src/serde_bin.rs:393]: Clippy said these were useless transmutes and suggested the new code

File: src/serde_bin.rs [hunk 2]
```diff
@@ -179,7 +179,7 @@ impl SerBin for u8 {
 
 impl SerBin for bool {
     fn ser_bin(&self, s: &mut Vec<u8>) {
-        s.push(if *self { 1 } else { 0 });
```
Comment (not-fl3) [inline src/serde_bin.rs:182]: I do not agree with clippy here

File: src/serde_json.rs [hunk 3]
```diff
@@ -815,7 +813,7 @@ where
 {
     fn ser_json(&self, d: usize, s: &mut SerJsonState) {
         s.out.push('[');
-        if self.len() > 0 {
```
Comment (not-fl3) [inline src/serde_json.rs:818]: and here

File: src/serde_json.rs [hunk 4]
```diff
@@ -983,7 +979,7 @@ where
     fn de_json(o: &mut DeJsonState, d: &mut Chars) -> Result<Self, DeJsonErr> {
         unsafe {
             let mut to = core::mem::MaybeUninit::<[T; N]>::uninit();
-            let top: *mut T = core::mem::transmute(&mut to);
```
Comment (not-fl3) [inline src/serde_json.rs:986]: and probably here

File: src/serde_json.rs [hunk 5]
```diff
@@ -1127,16 +1123,14 @@ where
     fn ser_json(&self, d: usize, s: &mut SerJsonState) {
         s.out.push('{');
         let len = self.len();
-        let mut index = 0;
-        for (k, v) in self {
```
Comment (not-fl3) [inline src/serde_json.rs:1131]: and here for sure

File: .github/workflows/rust.yml [hunk 6]
```diff
@@ -1,9 +1,48 @@
...
+          with:
+            toolchain: stable
+            target: ${{ matrix.config.target }}
+            override: true      
+        - uses: actions-rs/cargo@v1
+          with:
+            command: fmt
+            args: --check
+
+  clippy:
```
Comment (not-fl3) [inline .github/workflows/rust.yml:26]: maybe keep rustfmt, but keep clippy optional?

**Comment by knickish** [comment]
Removed the clippy CI and changes, kept the rustfmt ones

**Comment by not-fl3** [comment]
thanks!

**Pull request merged**